lua/wikis/commons/Squad/Auto.lua

```diff
@@ -0,0 +1,653 @@
+---@field references table<string, string>
+---@field wholeTeam boolean
+---@field position string
+---@field fromTeam string?
+---@field fromRole string?
+---@field toTeam string?
+---@field toRole string?
+---@field faction string?
+
+local DEFAULT_INCLUDED_ROLES = {
+	[SquadUtils.SquadType.PLAYER] = {
+		[SquadUtils.SquadStatus.ACTIVE] = {
+			'',
+			'Loan',
+			'Substitute',
+			'Trial',
+			'Stand-in',
+			'Uncontracted'
+		},
+		[SquadUtils.SquadStatus.INACTIVE] = {
+			'Inactive'
+		}
+	},
+	[SquadUtils.SquadType.STAFF] = {},
+}
+
+local DEFAULT_EXCLUDED_ROLES = {
+	[SquadUtils.SquadType.PLAYER] = {},
+	[SquadUtils.SquadType.STAFF] = {
+		'',
+		'Loan',
+		'Substitute',
+		'Trial',
+		'Stand-in',
+		'Uncontracted',
+		'Inactive'
+	},
+}
+
+---Entrypoint for SquadAuto tables
+---@param frame table
+function SquadAuto.run(frame)
+	local autosquad = SquadAuto(frame)
+	autosquad:parseConfig()
+	autosquad:queryTransfers()
+
+	local entries = autosquad:selectEntries()
+	Array.forEach(entries, FnUtil.curry(SquadAuto.enrichEntry, autosquad))
+
+	return autosquad:display(entries)
+end
+
+---Parses the args into a SquadAutoConfig
+function SquadAuto:parseConfig()
+	local args = self.args
+	local type = SquadUtils.TypeToSquadType[(args.type or ''):lower()]
+	local status = SquadUtils.StatusToSquadStatus[(args.status or ''):lower()]
+	self.config = {
+		team = args.team or mw.title.getCurrentTitle().text,
+		type = type,
+		status = status,
+		title = args.title,
+		roles = {
+			included = Logic.nilIfEmpty(Array.parseCommaSeparatedString(args.roles))
+				or DEFAULT_INCLUDED_ROLES[type][status],
+			excluded = Logic.nilIfEmpty(Array.parseCommaSeparatedString(args.not_roles))
+				or DEFAULT_EXCLUDED_ROLES[type],
+		}
+	}
+
+	self.manualPlayers, self.enrichmentInfo = self:readManualRowInput()
+
+	-- Override default 'Former Squad' title
+	if status == SquadUtils.SquadStatus.FORMER
+			and type == SquadUtils.SquadType.PLAYER
+			and not self.config.title then
+		self.config.title = 'Former Players'
+	end
+
+	local historicalTemplates = TeamTemplate.queryHistorical(self.config.team)
+	if not historicalTemplates then
+		error(TeamTemplate.noTeamMessage(self.config.team))
+	end
+	self.config.teams = Array.append(Array.extractValues(historicalTemplates), self.config.team)
+
+	if self.config.status == SquadUtils.SquadStatus.FORMER_INACTIVE then
+		error('SquadStatus \'FORMER_INACTIVE\' is not supported by SquadAuto.')
+	end
+
+end
+
+---@param entries SquadAutoPerson[]
+---@return Widget|Html|string?
+function SquadAuto:display(entries)
+	if Logic.isEmpty(entries) then
+		return
+	end
+
+	if self.config.status == SquadUtils.SquadStatus.FORMER then
+		return self:displayTabs(entries)
+	end
+
+	entries = SquadAuto._sortEntries(entries)
+
+	return SquadCustom.runAuto(entries, self.config.status, self.config.type, self.config.title)
+end
+
+---@param entries SquadAutoPerson[]
+---@return Html|string?
+function SquadAuto:displayTabs(entries)
```

Comment:
imo add a catch for the case that there was only 1 year and return without tabs in that case